sdk/python/feast/infra/gcp.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime
+from typing import List, Optional
+
+from feast import FeatureTable
+from feast.infra.provider import Provider
+from feast.repo_config import DatastoreOnlineStoreConfig
+
+
+def _delete_all_values(client, key) -> None:
+    """
+    Delete all keys under the key path in firestore.
+    """
+    while True:
+        query = client.query(kind="Value", ancestor=key)
+        entities = list(query.fetch(limit=1000))
+        if not entities:
+            return
+
+        for entity in entities:
+            print("Deleting: {}".format(entity))
+            client.delete(entity.key)
+
+
+class Gcp(Provider):
```

Comment:
Do we have a plan for reconciling the `OnlineStore` ABC and `OfflineStore` ABC approach that @jklegar is taking with the approach here (provider centric)?

Reply:
I'd see once both PRs land, I don't think they're in the way of each other right now but there may be a nicer structure